### ISSUE TITLE
Android app crash: Surface : getSlotFromBufferLocked: unknown buffer: 0xb812f398

### ISSUE THREAD

**Issue by aveltras** [obsidiansystems/obelisk#369]
I'm having trouble trying obelisk on Android right now.
The app produced crashes with the following log output (there might be lines coming from other processes than the app, i don't know how to filter these).
The crash happens instantly when i try to open the app on the phone.

```
03-12 16:20:03.706  1059  1655 I ActivityManager: Start proc 20283:systems.obsidian.obelisk.examples.minimal/u0a270 for activity systems.obsidian.obelisk.examples.minimal/systems.obsidian.HaskellActivity
03-12 16:20:03.707  1895  1895 I ContextualWidget: m_Listeners.size = 0
03-12 16:20:03.708 20283 20283 I art     : Late-enabling -Xcheck:jni
03-12 16:20:03.708 20283 20283 I art     : VMHOOK: rlim_cur : 0 pid:20283
03-12 16:20:03.712  1059  1945 W HtcSystemUPManager: HtcSystemUPHandler The policy is disabled. AppId: activity_history, category: launch
03-12 16:20:03.743  1059  1106 D StatusBarManagerService: setSystemUiVisibility(0x8600)
03-12 16:20:03.743  1059  1106 D StatusBarManagerService: disable:userId=0 what=0x0 which=0x1 pkg=Window{10eb53d u0 Starting systems.obsidian.obelisk.examples.minimal}
03-12 16:20:03.743  1059  1106 D StatusBarManagerService: hiding MENU key
03-12 16:20:03.743  1616  1616 I PhoneStatusBar: setSystemUiVisibility vis=8600 mask=ffffffff oldVal=8700 newVal=8600 diff=100
03-12 16:20:03.743  1616  1616 I PhoneStatusBar: updatePolicy(false,false,false,true)
03-12 16:20:03.743  1616  1616 I PhoneStatusBar: hiding the MENU button
03-12 16:20:03.747 20283 20283 W ActivityThread: Application systems.obsidian.obelisk.examples.minimal can be debugged on port 8100...
03-12 16:20:03.755  1895  2415 E Surface : getSlotFromBufferLocked: unknown buffer: 0xb812f398
03-12 16:20:03.763 20283 20283 W linker  : /data/app/systems.obsidian.obelisk.examples.minimal-1/lib/arm/libHaskellActivity.so: unused DT entry: type 0x1d arg 0xb49
03-12 16:20:03.780 20283 20296 D HaskellActivity: Logger running
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: unknown RTS option: -N2
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: Usage: <prog> <args> [+RTS <rtsopts> | -RTS <args>] ... --RTS <args>
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:    +RTS    Indicates run time system options follow
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:    -RTS    Indicates program arguments follow
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   --RTS    Indicates that ALL subsequent arguments will be given to the
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:            program (including any of these RTS flags)
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: The following run time system options are available:
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   -?       Prints this message and exits; the program is not executed
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   --info   Print information about the RTS used by this program
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   -K<size>  Sets the maximum stack size (default: 80% of the heap)
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:             Egs: -K32k -K512k -K8M
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   -ki<size> Sets the initial thread stack size (default 1k)  Egs: -ki4k -ki2m
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   -kc<size> Sets the stack chunk size (default 32k)
03-12 16:20:03.782 20283 20296 D HaskellActivity: jsaddle:   -kb<size> Sets the stack chunk buffer size (default 1k)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -A<size>  Sets the minimum allocation area size (default 1m) Egs: -A20m -A10k
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -AL<size> Sets the amount of large-object memory that can be allocated
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:             before a GC is triggered (default: the value of -A)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -n<size>  Allocation area chunk size (0 = disabled, default: 0)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -O<size>  Sets the minimum size of the old generation (default 1M)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -M<size>  Sets the maximum heap size (default unlimited)  Egs: -M256k -M1G
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -H<size>  Sets the minimum heap size (default 0M)   Egs: -H24m  -H1G
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -xb<addr> Sets the address from which a suitable start for the heap memory
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:             will be searched from. This is useful if the default address
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:             clashes with some third-party library.
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -m<n>     Minimum % of heap which must be available (default 3%)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -G<n>     Number of generations (default: 2)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -c<n>     Use in-place compaction instead of copying in the oldest generation
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:            when live data is at least <n>% of the maximum heap size set with
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:            -M (default: 30%)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -c       Use in-place compaction for all oldest generation collections
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:            (the default is to use copying)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -w       Use mark-region for the oldest generation (experimental)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -I<sec>  Perform full GC after <sec> idle time (default: 0.3, 0 == off)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -T         Collect GC statistics (useful for in-program statistics access)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -t[<file>] One-line GC statistics (if <file> omitted, uses stderr)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -s[<file>] Summary  GC statistics (if <file> omitted, uses stderr)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -S[<file>] Detailed GC statistics (if <file> omitted, uses stderr)
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.783 20283 20296 D HaskellActivity: jsaddle:   -Z         Don't squeeze out update frames on stack overflow
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -B         Sound the bell at the start of each garbage collection
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -h       Heap residency profile (output file <program>.hp)
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -i<sec>  Time between heap profile samples (seconds, default: 0.1)
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -C<secs>  Context-switch interval in seconds.
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             0 or no argument means switch as often as possible.
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             Default: 0.02 sec.
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -V<secs>  Master tick interval in seconds (0 == disable timer).
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             This sets the resolution for -C and the heap profile timer -i,
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             and is the frequence of time profile samples.
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             Default: 0.01 sec.
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   --install-signal-handlers=<yes|no>
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             Install signal handlers (default: yes)
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -e<n>     Maximum number of outstanding local sparks (default: 4096)
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -xq       The allocation limit given to a thread after it receives
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             an AllocationLimitExceeded exception. (default: 100k)
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:   -Mgrace=<n>
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             The amount of allocation after the program receives a
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             HeapOverflow exception before the exception is thrown again, if
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle:             the program is still exceeding the heap limit.
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.784 20283 20296 D HaskellActivity: jsaddle: RTS options may also be specified using the GHCRTS environment variable.
03-12 16:20:03.785 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.785 20283 20296 D HaskellActivity: jsaddle: Other RTS options may be available for programs compiled a different way.
03-12 16:20:03.785 20283 20296 D HaskellActivity: jsaddle: The GHC User's Guide has full details.
03-12 16:20:03.785 20283 20296 D HaskellActivity: jsaddle: 
03-12 16:20:03.785 20283 20297 D HaskellActivity: Haskell main action exited with status 1
03-12 16:20:03.811  1059  2087 D PMS     : acquireWL(8a9f132): PARTIAL_WAKE_LOCK  *launch* 0x1 1059 1000 WorkSource{10270}
03-12 16:20:03.817  1059  1945 W HtcSystemUPManager: HtcSystemUPHandler The policy is disabled. AppId: activity_history, category: launch
03-12 16:20:03.818  1059  1106 D StatusBarManagerService: setSystemUiVisibility(0x8700)
03-12 16:20:03.818  1059  1106 D StatusBarManagerService: disable:userId=0 what=0x0 which=0x1 pkg=Window{1e1b049 u0 com.htc.launcher/com.htc.launcher.Launcher}
03-12 16:20:03.819  1059  1106 D StatusBarManagerService: hiding MENU key
03-12 16:20:03.819  1616  1616 I PhoneStatusBar: setSystemUiVisibility vis=8700 mask=ffffffff oldVal=8600 newVal=8700 diff=100
03-12 16:20:03.820  1616  1616 I PhoneStatusBar: updatePolicy(false,false,false,true)
03-12 16:20:03.820  1616  1616 I PhoneStatusBar: hiding the MENU button
03-12 16:20:03.828  1895  1895 W PrismDragLayer: performFitSystemWindowsIfNeed: null, ic: null
03-12 16:20:03.829  1895  1895 I ContextualWidget: onResume
03-12 16:20:03.829  1895  1895 I ContextualWidget: notifyWidgetActive location size=0 user consent location=false
03-12 16:20:03.829  1895  1895 I ContextualWidget: ignore uploadUsageData location=false usage data=false  allowAutoUpload=true
03-12 16:20:03.829  1895  2289 I ContextualWidget: handleMessage, what=1018 mode=GettingOut maxcount=8
03-12 16:20:03.829  1895  1895 I ContextualWidget: postSyncRecommendedApps, isReady=false allowAutoSync=true syncMode=1 isEnableRecommend=true
03-12 16:20:03.830  1895  1895 W PrismThemeStaticWallpa_: checkWallpaperConsistency currentWallpaperHomeType is non-known:Single
03-12 16:20:03.831  1895  1895 I Launcher: LauncherPostResume
03-12 16:20:03.831  1895  2289 I PrismAccCustomizationU_: readDisableOOBEUserConsent: false
03-12 16:20:03.831  1895  1895 I FeedHostManagerProxy: setActivityPostResumed true
03-12 16:20:03.832  1059  2087 W InputMethodManagerService: Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@c634f56 attribute=null, token = android.os.BinderProxy@a163650
03-12 16:20:03.870  1059  1655 D PMS     : releaseWL(8a9f132): PARTIAL_WAKE_LOCK  *launch* 0x1 WorkSource{2897}
03-12 16:20:03.873 20283 20283 D Process : killProcess, pid=20283
03-12 16:20:03.874 20283 20283 D Process : systems.obsidian.HaskellActivity.onDestroy:127 android.app.Activity.performDestroy:6418 android.app.Instrumentation.callActivityOnDestroy:1160 
03-12 16:20:03.888  1059  1943 I ActivityManager: Recipient 20283
03-12 16:20:03.960  1059  1943 I ActivityManager: Process systems.obsidian.obelisk.examples.minimal (pid 20283) has died
```

I'm using a HTC One M8 (android v6.0.0).
According to [some stack overflow topic](https://stackoverflow.com/questions/33046375/e-surface-getslotfrombufferlocked-unknown-buffer-0xab7519c0), this is a problem relative to this specific Android version which has been fixed in v6.0.1 (which i can't upgrade to without flashing an unofficial ROM).
I wonder if something can be done to get around this on obelisk side or if it's a lost cause.

Thanks

**Comment by ryantrinkle** [comment]
@aveltras It looks like your android app isn't being built with `-threaded`. Can you post your cabal file(s)?

**Comment by aveltras** [comment]
I have reproduced the problem with the default skeleton just to be sure.
The cabal files are the default ones (-threaded option is there in the frontend.cabal for the executable).

**Comment by ryantrinkle** [comment]
Ah, ok; we'll take a look at that ASAP!

**Comment by kmicklas** [comment]
@aveltras I'm having a hard time finding this out on the internet, do you know if the HTC One M8 uses a 32 bit or 64 bit CPU?

**Comment by aveltras** [comment]
@kmicklas Mine is 32 bit according to AnTuTu Benchmark.

**Comment by kmicklas** [comment]
@aveltras In that case I think the crash is due to the error `jsaddle: unknown RTS option: -N2`, not the `getSlotFromBufferLocked` one. This looks to be a duplicate of https://github.com/reflex-frp/reflex-platform/issues/433 which we are working to resolve.

**Comment by kmicklas** [comment]
@aveltras This will be fixed by reflex-frp/reflex-platform#480. Once that's merged we'll update `reflex-platform` on Obelisk's `develop` branch.
